### PR TITLE
Added unit field to CashuWallet

### PR DESCRIPTION
**Pull request by starbackr-dev**
unit field is required for v1 API to quote, mint and melt tokens based on unit=usd or others. Updated CashuWallet and test scripts related to that.

# Fixes: #[issue]

## Description

...

## Changes
private _unit = 'sat'; Defaults to sat and optional
- ...
- ...

## PR Tasks

- [x] Open PR
- [x] run `npm run test` --> no failing unit tests
- [x] run `npm run format`

### CODE REVIEW

File: src/CashuWallet.ts [hunk 1]
```diff
@@ -40,16 +40,17 @@ import { wordlist } from '@scure/bip39/wordlists/english';
...
+	
 
 	/**
 	 * @param keys public keys from the mint
 	 * @param mint Cashu mint instance is used to make api calls
 	 * @param mnemonicOrSeed mnemonic phrase or Seed to initial derivation key for this wallets deterministic secrets. When the mnemonic is provided, the seed will be derived from it.
 	 * This can lead to poor performance, in which case the seed should be directly provided
 	 */
-	constructor(mint: CashuMint, keys?: MintKeys, mnemonicOrSeed?: string | Uint8Array) {
+	constructor(mint: CashuMint, unit?:string, keys?: MintKeys, mnemonicOrSeed?: string | Uint8Array) {
```
Comment (gandlafbtc) [inline src/CashuWallet.ts:53]: we pass in `unit` here, but do nothing with it? Why?

**Review by gandlafbtc — CHANGES_REQUESTED** [review]
Can you take another look please?

File: src/CashuWallet.ts (hunk 1 above)
Comment (starbackr-dev) [reply src/CashuWallet.ts:53]: working on it. I think the merge didn't port the changes. Will test and push it.

Comment (starbackr-dev) [reply src/CashuWallet.ts:53]: ok. fixed it. To test try this. 

//For creating invoice using Strike backend
const wallet = new CashuWallet(new CashuMint('https://stablenut.umint.cash'), 'usd');

//For creating invoice using LND backend
const wallet = new CashuWallet(new CashuMint('https://stablenut.umint.cash'));

File: src/CashuWallet.ts [hunk 2]
```diff
@@ -40,17 +40,19 @@ import { wordlist } from '@scure/bip39/wordlists/english';
...
```
Comment (gandlafbtc) [inline src/CashuWallet.ts:53]: Should we also use an `options?` object param, and put the optional parameters in that object?

**Review by gandlafbtc — APPROVED** [review]
LGTM, One minor thing, but not related to this PR, so I think we can add it later

**Review by gandlafbtc — APPROVED** [review]

**Pull request merged**